### DISCUSSION TITLE
Apple Store postprocessing fails

### DISCUSSION THREAD

**Discussion by abdulkash** [codemagic-ci-cd/codemagic-docs#2213] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

After build the error occurs during postprocessing for apple appstore

### Expected results

Succcessful app submission for review

### Actual results

Failure: POST https://api.appstoreconnect.apple.com/v1/reviewSubmissionItems returned 409: appStoreVersions with id '30a8cd89-3bc2-4d01-92a9-b4ac711777c6' is not in valid state. - This resource cannot be reviewed, please check associated errors to see why.
	Associated error: The provided entity is missing a required attribute - You must provide a value for the attribute 'whatsNew' with this request
	Associated error: The provided entity is missing a required attribute - You must provide a value for the attribute 'usesNonExemptEncryption' with this request

**Comment by himesh-cm** [comment]
Hi @abdulkash,
Have you specified the `whatsNew` in your `release notes` file?
Please share a build id, so that i can further investigate the issue.

**Reply by abdulkash** [reply]
I haven't added a release_note file yet

**Reply by himesh-cm** [reply]
@abdulkash - Please share a build id, so that i can further investigate the issue.

**Comment by icarusdust** [comment]
Hi, I will close this thread due to inactivity but before that let me highlight a couple of things. 

If you are facing this issue at the Post-processing step, then it would mean that:

 1. There is a `release_notes.json` file found in the root directory of your repo
 2. You are submitting the build to TestFlight beta review.

The post-processing step is triggered in these cases, not particularly when submitting your build to App Store Connect as build submission is already resolved at the Publishing phase. So, it could be either of the above in your case, and worth mentioning that release notes could be generated from your commit messages as well where the required fields are missing. For your self-investigation, you can check it under the Publishing step:
```
== Gathering artifacts ==
Reading release notes from "release_notes.json"
== Publishing artifacts ==
```
It would mean that the release notes are retrieved, and missing the required field in your case for ex.

**Reply by omar-chaabouni** [reply]
Thank you, I just wanted to add documentation link for anyone in need : https://docs.codemagic.io/yaml-notification/publish-release-notes/
